sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingInterpreter.scala

```diff
@@ -124,6 +124,22 @@ case class SqlScriptingInterpreter() {
         val bodyExec = transformTreeIntoExecutable(body, session).asInstanceOf[CompoundBodyExec]
         new LoopStatementExec(bodyExec, label)
 
+      case ForStatement(query, variableNameOpt, body, label) =>
+        val queryExec = new SingleStatementExec(query.parsedPlan, query.origin, isInternal = false)
+        val bodyExec =
+          transformTreeIntoExecutable(body, session).asInstanceOf[CompoundBodyExec]
+        val finalExec = variableNameOpt match {
+          case None => bodyExec
+          case Some(variableName) =>
+            val dropVariableExec = new SingleStatementExec(
+              DropVariable(UnresolvedIdentifier(Seq(variableName)), ifExists = true),
+              Origin(),
+              isInternal = true
+            )
+            new CompoundBodyExec(Seq(bodyExec, dropVariableExec))
```

Comment:
can't we just append `dropVariableExec` to the `collection` within `bodyExec`?
current approach seems like unnecessary nesting, which we should avoid if possible.
it would also make it equivalent to what we are doing for regular compounds.

Reply:
Sure, will change. For some reason i thought we did it like this for compounds.